### PR TITLE
Update k6 dependency and JS module structure to v0.35.0

### LINKED ISSUE #121: Unabel to build xk6-browser
```
go version
go version go1.17.3 darwin/amd64
```

```
sw_vers
ProductName:	macOS
ProductVersion:	11.5.2
BuildVersion:	20G95
```
```
xk6 build --output xk6-browser --with github.com/grafana/xk6-browser@main
2021/11/17 16:49:39 [INFO] Temporary folder: /Users/untitled/buildenv_2021-11-17-1649.864137876
2021/11/17 16:49:39 [INFO] Writing main module: /Users/untitled/buildenv_2021-11-17-1649.864137876/main.go
2021/11/17 16:49:39 [INFO] Initializing Go module
2021/11/17 16:49:39 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init k6
go: creating new go.mod: module k6
go: to add module requirements and sums:
	go mod tidy
2021/11/17 16:49:39 [INFO] Pinning versions
2021/11/17 16:49:39 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod edit -require go.k6.io/k6@latest
2021/11/17 16:49:39 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy
go: finding module for package github.com/grafana/xk6-browser
go: found github.com/grafana/xk6-browser in github.com/grafana/xk6-browser v0.1.0
2021/11/17 16:49:40 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod edit -require github.com/grafana/xk6-browser@main
2021/11/17 16:49:40 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy
2021/11/17 16:49:41 [INFO] Build environment ready
2021/11/17 16:49:41 [INFO] Building k6
2021/11/17 16:49:41 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy
2021/11/17 16:49:41 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /Users/untitled/xk6-browser -ldflags -w -s -trimpath
# github.com/grafana/xk6-browser
../go/pkg/mod/github.com/grafana/xk6-browser@v0.1.1-0.20211116085927-baaf58caef73/main.go:43:3: undefined: modules.InstanceCore
../go/pkg/mod/github.com/grafana/xk6-browser@v0.1.1-0.20211116085927-baaf58caef73/main.go:50:3: undefined: modules.InstanceCore
../go/pkg/mod/github.com/grafana/xk6-browser@v0.1.1-0.20211116085927-baaf58caef73/main.go:56:4: undefined: modules.IsModuleV2
../go/pkg/mod/github.com/grafana/xk6-browser@v0.1.1-0.20211116085927-baaf58caef73/main.go:67:40: undefined: modules.InstanceCore
2021/11/17 16:49:42 [INFO] Cleaning up temporary folder: /Users/untitled/buildenv_2021-11-17-1649.864137876
2021/11/17 16:49:42 [FATAL] exit status 2
````

For example xk6-notification has been built fine

### PR DESCRIPTION
**Pull request by imiric**
Closes #121

### CODE REVIEW

**Review by robingustafsson — APPROVED** [review]
Code LGTM, but I haven't dug into the new module API yet. I do think we eventually should switch to exporting the `BrowserType`s explicitly like Playwright:

```js
import { chromium, firefox, webkit } from "k6/x/browser";

const browser = chromium.launch(opts);
```
rather than our current:
```js
import launcher from "k6/x/browser";

const browser = launcher.launch('chromium', opts);
```
This PR also reminded me that we need to fix our ["devices"](https://github.com/grafana/xk6-browser/blob/main/common/device.go) list so you'd ideally be able to write:
```js
import { chromium, devices } from "k6/x/browser";

const browser = chromium.launch(opts);
const context = browser.newContext({ ...devices['iPhone 13'] });
```
I'll create an issue about it...

File: main.go [hunk 1]
```diff
@@ -64,20 +63,19 @@ func New() *RootModule {
...
-			Devices:      common.GetDevices(),
-			Version:      version,
+			vu:      vu,
+			Devices: common.GetDevices(),
+			Version: version,
 		},
 	}
 }
 
 // GetExports returns the exports of the JS module so that it can be used in
```
Comment (inancgumus) [inline main.go:76]: The comment `GetExports` -> `Exports`

File: main.go [hunk 2]
```diff
@@ -64,20 +63,19 @@ func New() *RootModule {
 
 // NewModuleInstance implements the modules.IsModuleV2 interface to return
```
Comment (inancgumus) [inline main.go:64]: The comment `modules.IsModuleV2` to `modules.Module`.

**Review by inancgumus — CHANGES_REQUESTED** [review]
LGTM, we just need to fix the comments.

File: main.go (hunk 1 above)
Comment (imiric) [reply main.go]: :facepalm: Thanks, we should really prioritize #58.

**Review by inancgumus — APPROVED** [review]

**Pull request merged**